app/stores/pokemon.js

```diff
@@ -129,6 +129,26 @@ module.exports = {
         }.bind(this));
     },
     /*
+     * get the pokemon details of particular pokemon based on description
+     */
+    getByDescription : function (description, callback) {
+        this.get({'description': new RegExp('^.*' + description + '.*$', 'i')}, function (status, response) {
+            if (status === 1 && _.isEmpty(response) && description.length > 8) {
+                this.getAll(function (status, response) {
+                    if (status === 1) {
+                        this.getClosest(description, response, function (status, response) {
+                            callback(status, response);
+                        });
+                    } else {
+                        callback(status, response);
+                    }
+                }.bind(this));
+            } else {
+                callback(status, response);
+            }
+        }.bind(this));
+    },
+    /*
```

Comment:
Is it required to do a partial search based on hamming algorithm for description too?  Maybe just the regex based search is sufficient.